### ISSUE TITLE
[BUG] Boost.Hana not found only at -O3

### ISSUE THREAD

**Issue by tweej** [mattgodbolt/compiler-explorer#1656]
**Describe the bug**
I can get code that uses Boost.Hana compiling with x86-64 gcc 9.2 at -O0, -O1, and -O2 without any problems: https://godbolt.org/z/bkQ55N

Changing the optimization level to -O3 results in the error:
```
In file included from <source>:1:

/celibs/boost_1_71_0/boost/hana.hpp:45:10: fatal error: /celibs/boost_1_71_0/boost/hana/config.hpp: Permission denied

   45 | #include <boost/hana/config.hpp>

      |          ^~~~~~~~~~~~~~~~~~~~~~~

compilation terminated.

Compiler returned: 1
```

https://godbolt.org/z/2vuTT4

**To Reproduce**
Steps to reproduce the behavior: https://godbolt.org/z/2vuTT4

**Expected behavior**
The program should compile at -O3 the same way it works at -O2.

**Link**
https://godbolt.org/z/2vuTT4

**Screenshots**
None

**Desktop (please complete the following information):**
 - OS: Ubuntu 18.04
 - Browser Google Chrome
 - Version 78.0.3904.70 (Official Build) (64-bit)

**Additional context**
None

**Comment by apmorton** [comment]
You have been bitten by caching of a failed compile here - in the future if this kind of error happens with boost libraries please click the "clear cache and recompile" button in the output window - shown below.
![image](https://user-images.githubusercontent.com/63636/68053602-b7dbc880-fcc2-11e9-9133-c5986ea82335.png)

Some background on exactly what is going on here if you are interested #1303 

TL;DR; is that when new instances start up they have to sync the boost library (specifically, not any others) because boost is so massive - during this process they can return failed compiles and unfortunately these will be cached.

**Comment by tweej** [comment]
👍 Thanks.
